### DISCUSSION TITLE
[RAC]component cannot be used as a JSX component according to tsc

### DISCUSSION THREAD

**Discussion by synecdokey** [adobe/react-spectrum#5161] [Q&A]
Whenever I try using a component from `react-aria-components`, the typescript compiler tells me some version of this error:

```
src/components/Header/index.tsx:18:8 - error TS2786: 'Button' cannot be used as a JSX component.
  Its type '(props: ButtonProps & RefAttributes<HTMLButtonElement>) => ReactElement<any, string | JSXElementConstructor<any>>' is not a valid JSX element type.
    Type '(props: ButtonProps & RefAttributes<HTMLButtonElement>) => ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type 'ReactNode'.
```

It runs fine at runtime, it's only an issue for Typescript.

I am using the beta version of `react-aria-components`, my typescript is version `5.1`, and my `@types/react` is `18.2.21`.

**Comment by reidbarber** [comment]
This may be related to the active discussion in https://github.com/adobe/react-spectrum/issues/5157

**Comment by snowystinger** [comment]
Would you mind trying upgrading your TS and types/react to the latest versions? also check if you have multiple versions of @types/react
